xcdat/dataset.py

```diff
@@ -0,0 +1,102 @@
+"""Dataset module for Dataset related functions."""
+import pandas as pd
+import xarray as xr
+
+from xcdat import axis  # noqa: F401
+
+
+def open_dataset(path: str) -> xr.Dataset:
+    """Wrapper for xarray's ``open_dataset()`` to apply common operations.
+
+    Operations include:
+
+    - Decode all time units, including non-CF compliant units (months and years)
+    - Generate lat/lon coordinate bounds if they don't exist
+
+    Parameters
+    ----------
+    path : str
+        Path to Dataset.
+
+    Returns
+    -------
+    xr.Dataset
+        Dataset after applying operations.
+
+    Notes
+    -----
+    xarray's ``open_dataset`` opens the file with read-only access. When you
+    modify values of a Dataset, even one linked to files on disk, only the
+    in-memory copy you are manipulating in xarray is modified: the original file
+    on disk is never touched.
+
+    Examples
+    --------
+    Import and call module:
+
+    >>> from xcdat import open_dataset
+    >>> ds = open_dataset("file_path")
+    """
+    ds = xr.open_dataset(path, decode_times=False)
```

Comment:
@tomvothecoder - One xarray feature that I expect we will frequently rely on is the [open_mfdataset](http://xarray.pydata.org/en/stable/generated/xarray.open_mfdataset.html) functionality. I'm wondering if we could automatically decide whether to use `open_dataset` or `open_mfdataset` by checking if the `path` parameter is a file or a directory, e.g., something like this:

```
import os
...
if os.path.isdir(path):
    ds = xr.open_mfdataset(path + '/*')
elif os.path.isfile(path):
    ds = xr.open_dataset(path)
else:
    raise ValueError('Input path (' + path + ') is invalid.')
```

Reply:
This might require adding dask as a [dependency](https://github.com/tomvothecoder/xcdat/blob/main/conda-recipe/meta.yaml)

Reply:
Thanks for bringing this up @pochedls.

I need to familiarize myself with `open_mfdataset()` to see if there are any caveats/complexities in regards to generating bounds and fixing time units. I'm assuming we'd use this function to join multiple files within a single dataset as a single xarray `Dataset` object, which means they share the same coordinates and bounds?

Reply:
@tomvothecoder - yes you are exactly right. A large fraction of climate data spans multiple netCDF files (each file typically has a chunk of time, e.g., 1970 - 1979, 1980 - 1989, etc.).